operators/src/source/gdal_source.rs

```diff
@@ -400,14 +413,30 @@ where
                 &time_string.unwrap_or_else(|| "".into()),
             );
 
+        dbg!(&file_name);
+
         let path = gdal_dataset_information.dataset_path(); // TODO: add the path of the definition file for relative paths
         let data_file = path.join(file_name);
 
-        // open the dataset at path (or 'throw' an error)
-        let dataset = GdalDataset::open(&data_file)?; // TODO: investigate if we need a dataset cache
-                                                      // get the geo transform (pixel size ...) of the dataset (or 'throw' an error)
-                                                      // let gdal_geo_transform = dataset.geo_transform()?;
-                                                      // let geo_transform = GeoTransform::from(gdal_geo_transform); //TODO: clip the geotransform information / is this required at all?
+        dbg!(&data_file);
+
+        let tile_grid = tile_information.tile_size_in_pixels();
+
+        // open the dataset at path
+        let dataset_result = GdalDataset::open(&data_file);
+        // TODO: investigate if we need a dataset cache
+
+        // shortcut if there is raster file -> return a no-data file.
+        if dataset_result.is_err() {
+            return Ok(RasterTile2D::new_with_tile_info(
+                time_interval,
+                tile_information,
+                Grid2D::new_filled(tile_grid, T::zero(), None),
+            ));
+        };
+
+        // this was checked one line above...
+        let dataset = dataset_result.expect("checked");
```

Comment:
You could say 
```
let dataset = if let Ok(dataset) = dataset_result {
    dataset_result
} else {
    return …;  // NODATA tile
};
```
to circumvent the double check.

Reply:
yes but the block is so large i really don't want to do this.